Documentation changes

aogriffiths opened the pull request:
I'd like to contribute to HassOS and thought I'd start with some documentation changes. Please consider this pull request for a new README.md, and a spelling correction to kernal.md. 

Thanks
Adam

homeassistant commented:
Hi @aogriffiths, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;operating-system#768).

Once you do that we will be able to review and accept this pull request.

Thanks!

aogriffiths commented:
I'fe now signed the CLA

frenck commented on README.md:
```diff
@@ -1,18 +1,62 @@
-# Home Assistant Operating system
+# Home Assistant Operating System
```
New line after first heading

```suggestion
# Home Assistant Operating System

```

aogriffiths replied:
Done

frenck commented on README.md:
```diff
@@ -1,18 +1,62 @@
 
-This is an embedded Linux which works different than a normal Linux distribution. The system is designed to run with minimal I/O and is optimized for its tasks.
+## Features
+* Lightweight and memory efficient
+* Minimized I/O
+* Over The Air (OTA) updates
+* Offline updates
+* Modular using Docker
```
- Use dashes for unordered lists.
- Headings should be surrounded by newlines
- `memory-efficient`

```suggestion

- Lightweight and memory-efficient
- Minimized I/O
- Over The Air (OTA) updates
- Offline updates
- Modular using Docker
```

frenck commented on README.md:
```diff
@@ -1,18 +1,62 @@
+* Modular using Docker
 
-If you don't have experience with embedded systems, buildroot or the build process Linux distributions, then please read up on those topics. All provided documentation here is focusing on developers with a background on embedded systems or a strong understanding of the internal workings of operating systems.
+## Supported hardware
+* Raspberry Pi
+* Hardkernel Odroid
+* Intel NUC
+* Asus Tinker Board
+* Virtual appliances
```
- Surround headings with newlines
- `ODROID`
- Use dashes for unordered lists

```suggestion
## Supported hardware

- Raspberry Pi
- Hardkernel ODROID
- Intel NUC
- Asus Tinker Board
- Virtual appliances
```

frenck commented on README.md:
```diff
@@ -1,18 +1,62 @@
-- Docker-CE
-- AppArmor protected
-- ZRAM LZ4 for `/tmp`, `/var` and swap
+## Getting Started
+If you just want to use Home Assistant the official [getting started guide](https://www.home-assistant.io/getting-started/) and [installation instructions](https://www.home-assistant.io/hassio/installation/) take you through how to download HassOS and get it running on your machine.
+
+If you're interested in finding out more about HassOS and how it works  read on...
```
```suggestion
## Getting Started

If you just want to use Home Assistant the official [getting started guide](https://www.home-assistant.io/getting-started/) and [installation instructions](https://www.home-assistant.io/hassio/installation/) take you through how to download HassOS and get it running on your machine.

If you're interested in finding out more about HassOS and how it works read on...
```

frenck commented on README.md:
```diff
@@ -1,18 +1,62 @@
+
+If you're interested in finding out more about HassOS and how it works  read on...
+
+## HassOS components
+
+
+* **Bootloader:**
+  * [Barebox](https://barebox.org/) for devices that support EFI
+  * [U-Boot](https://www.denx.de/wiki/U-Boot) for devices that don't support EFI
+* **Operating System:**
+  * [Buildroot](https://buildroot.org/) LTS Linux
+* **File Systems:**
+  * [SquashFS](https://www.kernel.org/doc/Documentation/filesystems/squashfs.txt) for read-only file systems (using LZ4 compression)
+  * [ZRAM](https://www.kernel.org/doc/Documentation/blockdev/zram.txt) for `/tmp`, `/var` and swap (using LZ4 compression)
+* **Container Platform:**
+  * [Docker Engine](https://docs.docker.com/engine/) for running Home Assistant components in containers
+* **Updates:**
+  * [RAUC](https://rauc.io/) for Over The Air (OTA) and USB updates
+* **Security:**
+  * [AppArmor](https://apparmor.net/) Linux kernel security module
```
```suggestion
## HassOS components

- **Bootloader:**
  - [Barebox](https://barebox.org/) for devices that support EFI
  - [U-Boot](https://www.denx.de/wiki/U-Boot) for devices that don't support EFI
- **Operating System:**
  - [Buildroot](https://buildroot.org/) LTS Linux
- **File Systems:**
  - [SquashFS](https://www.kernel.org/doc/Documentation/filesystems/squashfs.txt) for read-only file systems (using LZ4 compression)
  - [ZRAM](https://www.kernel.org/doc/Documentation/blockdev/zram.txt) for `/tmp`, `/var` and swap (using LZ4 compression)
- **Container Platform:**
  - [Docker Engine](https://docs.docker.com/engine/) for running Home Assistant components in containers
- **Updates:**
  - [RAUC](https://rauc.io/) for Over The Air (OTA) and USB updates
- **Security:**
  - [AppArmor](https://apparmor.net/) Linux kernel security module
```

frenck commented on README.md:
```diff
@@ -1,18 +1,62 @@
+
+If you don't have experience with these, embedded systems, buildroot or the build process for Linux distributions, then please read up on these topics. The rest of the documentation in this project is for developers and assumes you have experience of embedded systems or a strong understanding of the internal workings of operating systems.
+
+## Developer Documentation
+
+
+
+* [Getting started](./Documentation/getting_started_development.md) - the place for developers to begin
+* [Development](./Documentation/getting_started_development.md) - more deatils for developers
+* [Deployment](./Documentation/deployment.md) - approach to git branching and releases
+* [Configuration](./Documentation/configuration.md) - how users can configure HassOS
+* [Partition](./Documentation/partition.md) - partition layout
+* [Network](./Documentation/network.md) - approach to networking
+* [Bluetooth](./Documentation/bluetooth.md) - approach to bluetooth
+* [Kernel](./Documentation/kernel.md) - kernel versions
+
+
+
+(This documentation is kept in the [Documentation](./Documentation) directory.)
```
```suggestion
## Developer Documentation

- [Getting started](./Documentation/getting_started_development.md) - the place for developers to begin
- [Development](./Documentation/getting_started_development.md) - more deatils for developers
- [Deployment](./Documentation/deployment.md) - approach to git branching and releases
- [Configuration](./Documentation/configuration.md) - how users can configure HassOS
- [Partition](./Documentation/partition.md) - partition layout
- [Network](./Documentation/network.md) - approach to networking
- [Bluetooth](./Documentation/bluetooth.md) - approach to bluetooth
- [Kernel](./Documentation/kernel.md) - kernel versions

(This documentation is kept in the [Documentation](./Documentation) directory.)
```

pvizeli commented on Documentation/kernel.md:
```diff
@@ -5,7 +5,7 @@ Default Kernel tree: 5.4
 | Board | Version |
 |-------|---------|
-| Open Virtual Applicance | 5.4.50 |
+| Open Virtual Appliance | 5.4.50 |
```
This file is automatically updated, you need fix in the script too, otherwise the update process fails

aogriffiths replied:
Resolved, see https://github.com/home-assistant/operating-system/pull/768/commits/0162cbb722d1191dcca644d9a1079900318b0689

aogriffiths commented:
Hi all, thanks for all your comments. I'll take a look tomorrow and merge in to my pull request.

aogriffiths commented:
Hi @frenck and @pvizeli, I have followed all your requests on this PR so hopefully it's ready to go. I have also done further documentation updates in a new PR here https://github.com/home-assistant/operating-system/pull/769.

fabaff commented on README.md:
```diff
@@ -1,18 +1,61 @@
+- Asus Tinker Board
+- Virtual appliances
+
+See full list and specific models [here]
```
Where should that link go?

aogriffiths replied:
Will fix

aogriffiths replied:
resolved in commit 0cec61fba573fedbae0fb311bf556125075bc150

fabaff commented on README.md:
```diff
@@ -1,18 +1,61 @@
+Home Assistant Operating System (HassOS) is an operating system optimized for hosting [Home Assistant](https://www.home-assistant.io) and its [Add-ons](https://www.home-assistant.io/addons/).
 
-This is an embedded Linux which works different than a normal Linux distribution. The system is designed to run with minimal I/O and is optimized for its tasks.
+HassOS is uses Linux as a hypervisor for Docker, allowing each component of Home Assistant including add-ons to run in separate docker containers. It uses embedded Linux based on [buildroot](https://buildroot.org/), which is different to a "normal" Linux, and it is optimized for running Home Assistant on IoT devices.
```
> ...and it is optimized for running Home Assistant on IoT devices.

Hmmm, Home Assistant doesn't run on IoT devices per se. SBC can be IoT devices but you can't run Home Assistant on a temperature sensor.

> which is different to a "normal" Linux

It's different to a classical Linux distribution. Still Linux.

aogriffiths replied:
Agreed. I’d left these words in since they were from the original author of the readme. I’ll change them based on your suggestions.

aogriffiths replied:
Resolved in commit e1cc5f461d35c1881c41122352d11fe0d184d07e

aogriffiths commented:
Hi @frenck, @pvizeli, @fabaff, I have addressed all your comments for this PR. It would be great if you can approve this PR and I'll then move on to making more documentation improvements as part of https://github.com/home-assistant/operating-system/pull/769

aogriffiths commented:
Note the conflict on Documentation/kernel.md, Appliance vs. Applicance, should be resolved in favour of the correct spelling Appliance.  I have updated the script based on @pvizeli's comment above (https://github.com/home-assistant/operating-system/pull/768#discussion_r453365947)

aogriffiths commented:
closing this PR since #769 got  merged in.
